### PR TITLE
Updated arithmetic operations between `Tensor` and `Number`

### PR DESCRIPTION
**Pull request by eyjafjallac**
- Updated the Operable class in `abstracts.py` to include reverse operator methods for scalar operations.
- Added new operator functions in `tensors.py` to support element-wise operations between tensors and scalars, including multiplication, and division.
-  Added addition and subtraction with the special broadcasting rule while Rank 3 tensor is operating with the scalar.
- Implemented tests in `test_tensors.py` to validate scalar-tensor operations and ensure correct behavior, including edge cases for rank-1 tensors.

### CODE REVIEW

File: src/pyhilbert/tensors.py [hunk 1]
```diff
@@ -707,3 +855,20 @@ def expand_to_union(tensor: Tensor, union_dims: list[StateSpace]) -> Tensor:
         return tensor
 
     return Tensor(data=tensor.data.expand(target_shape), dims=tuple(new_dims))
+
+
+def identity(dims: Tuple[StateSpace, ...], device=None, dtype=None) -> Tensor:
```
[the remark this answers is not in the document]
Comment (eyjafjallac) [reply src/pyhilbert/tensors.py]: The device will follow the tensor itself (while doing addition or subtraction), I think user would not able to use it, except they want to form the identity matrix?

**Review by Alphaharrius — APPROVED** [review]
